### PR TITLE
ViewComponent sidecar

### PR DESCRIPTION
**Pull request by dacook**
#### What? Why?
To encapsulate javascript for ViewComponents in the same folder as the HTML and CSS

I wasn't actually happy with it at first and was just going to share as an FYI. But now I'm getting used to it. 
What do others think?

Reference: https://viewcomponent.org/guide/javascript_and_css.html

#### What should we test?
specs pass

### CODE REVIEW

**Review by mkllnk — CHANGES_REQUESTED** [review]
Interesting, indeed. I don't know yet if I like it. The concept of components, bundling it all together seems nice. But as you say, the implementation has some downsides. Time will tell, happy to give this a go.

The specs need updating though.

File: app/components/vertical_ellipsis_menu_component/component_controller.js
[the remark this answers is not in the document]
Comment (dacook) [reply app/components/vertical_ellipsis_menu_component/component_controller.js:1]: Magic! 🧙‍♂️
https://stimulus.hotwired.dev/handbook/installing#controller-filenames-map-to-identifiers

File: app/components/vertical_ellipsis_menu/component.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+class VerticalEllipsisMenu::Component < ViewComponent::Base
```
[the remark this answers is not in the document]
Comment (dacook) [reply app/components/vertical_ellipsis_menu/component.rb]: oopsie!

**Review by dacook — COMMENTED** [review]
Thanks, have fixed that.

**Review by mkllnk — APPROVED** [review]

**Review by rioug — APPROVED** [review]
I like having the component related files together it makes it easier to find things. They tend to do that in the React world, but they even go a step further by having the spec/test files in the component folder.

**Pull request merged**